searx/engines/duckduckgo.py

```diff
@@ -39,13 +37,22 @@ def request(query, params):
     offset = (params['pageno'] - 1) * 30
 
     if params['language'] == 'all':
-        locale = 'en-us'
+        locale = None
     else:
-        locale = params['language'].replace('_', '-').lower()
-
-    params['url'] = url.format(
-        query=urlencode({'q': query, 'kl': locale}),
-        offset=offset)
+        locale = params['language'].split('_')
+        if len(locale) == 2:
+            # country code goes first
+            locale = locale[1].lower() + '-' + locale[0].lower()
+        else:
+            # doesn't actually do anything because ddg requires both country and language
```

Comment:
I tried your PR with a locale that is only one long. It did not work as you had written here. If DDG requires both country and language why did you include the ``else``? Is it possible that there will be locales which match the format DDG requires without any transformation?